ingestion/src/metadata/profiler/metrics/static/distinct_count.py

```diff
@@ -41,15 +42,30 @@ def metric_type(self):
 
     @_label
     def fn(self):
+        """
+        Distinct Count metric for Sqlalchemy connectors
+        """
         return func.count(distinct(CountFn(column(self.col.name, self.col.type))))
 
     def df_fn(self, dfs=None):
+        """
+        Distinct Count metric for Datalake
+        """
         from collections import Counter  # pylint: disable=import-outside-toplevel
 
         try:
             counter = Counter()
             for df in dfs:
-                counter.update(df[self.col.name].dropna().to_list())
+                col_type = df[self.col.name].dtypes.name
+                if (
+                    col_type == "object"
+                    and self.col.type in [DataType.ARRAY, DataType.JSON]
+                    and any(df[self.col.name].values)
```

Comment:
I think you could do something like `and df[self.col.name].values` if you are checking if a column has no values

Reply:
this throws an error when the values are array
throws:
The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

Reply:
whereas any works for all scenarios that we want